### ISSUE TITLE
[Documentation]: Add automation to add release notes to release_notes.md file

### ISSUE THREAD

**Issue by aliabbasjaffri** [mlinfra-io/mlinfra#102]
### Describe the documentation change

Currently release notes are lost somewhere after a release is published. We would want to capture those release notes and append them to [release_notes.md file](https://github.com/mlinfra-io/mlinfra/blob/main/RELEASE_NOTES.md) via an automation. See [here](https://github.com/oprypin/mkdocs-gen-files) for more details.

### Importance of the change

Currently the release notes file is not getting populated with all the release notes from a release publish, and this information gets lost somewhere on github. Having a consistent release log of changes introduces more transparency in the system and allows users to easily see which version has certain features.

**Comment by Brijeshthummar02** [comment]
@aliabbasjaffri i can do this, assign me.

**Comment by aliabbasjaffri** [comment]
@Brijeshthummar02 assigned! do let me know if you have any questions or want to understand the requirement :-)

**Comment by Brijeshthummar02** [comment]
Can you show me the path to repo.

**Comment by aliabbasjaffri** [comment]
Which path are you referring to here?

**Comment by pearguacamole** [comment]
@aliabbasjaffri  I'd like to work on this, can you assign this to me also.

**Comment by aliabbasjaffri** [comment]
@Brijeshthummar02 are you working on this task?

**Comment by aliabbasjaffri** [comment]
@pearguacamole assigned you the issue. Let me know if you have any questions

**Comment by pearguacamole** [comment]
Hi, I have added a step in python-publish.yml to automatically append release notes into [release_notes.md file](https://github.com/mlinfra-io/mlinfra/blob/main/RELEASE_NOTES.md) when a new release is published. I have tested that it is working while manually running the workflow and is also being triggered when a new release is published, and the release_notes.md file is being populated. Only thing is that the complete workflow of [Upload python package](https://github.com/mlinfra-io/mlinfra/actions/workflows/python-publish.yml) is not being executed, as its not setup for my fork. Should I create a pr now as the step added should not affect working of any other part.

**Comment by aliabbasjaffri** [comment]
Hey @pearguacamole thank you so much for your effort on this task. The implementation looks good, however i was wondering what made you go for a bash script instead of a python component that does the job and its called from the step?

**Comment by pearguacamole** [comment]
Hi @aliabbasjaffri, So the task can definitely be done using a python component, which is called from the step. But the task is too specific to warrant use of a separate python component, if we use python we can reduce 3-4 lines of code (13 -> 8-9) but will have a new python file to maintain and there is no real use of the component anywhere else in this case. Also a separate component will reduce readability of the code.
Basically I think bash script is a bit cleaner and more readable compared to using a separate python component. It will also be easier to maintain. Hope that makes sense :-)

**Comment by aliabbasjaffri** [comment]
Awesome! I left a comment on the PR, otherwise it looks good to me. :-)
